scripts/exregional_run_analysis.sh

```diff
@@ -643,7 +705,130 @@ if [ $netcdf_diag = ".true." ]; then
+#
+# comment out for testing
+$APRUN ./gsi.x < gsiparm.anl > stdout_mem$memid 2>&1 || print_err_msg_exit "\
+Call to executable to run GSI returned with nonzero exit code."
 
+mv gsiparm.anl gsiparm.anl_mem$memid
+ls -l * > list_run_directory_mem$memid
+
+#-----------------------------------------------------------------------
+# Loop over first and last outer loops to generate innovation
+# diagnostic files for indicated observation types (groups)
+#
+# NOTE:  Since we set miter=2 in GSI namelist SETUP, outer
+#        loop 03 will contain innovations with respect to 
+#        the analysis.  Creation of o-a innovation files
+#        is triggered by write_diag(3)=.true.  The setting
+#        write_diag(1)=.true. turns on creation of o-g
+#        innovation files.
+#-----------------------------------------------------------------------
+#
+
+netcdf_diag=${netcdf_diag:-".false."}
+binary_diag=${binary_diag:-".true."}
+
+loops="01 03"
+for loop in $loops; do
+
+case $loop in
+  01) string=ges;;
+  03) string=anl;;
+   *) string=$loop;;
+esac
+
+#  Collect diagnostic files for obs types (groups) below
+if [ $binary_diag = ".true." ]; then
+   listall="conv hirs2_n14 msu_n14 sndr_g08 sndr_g11 sndr_g11 sndr_g12 sndr_g13 sndr_g08_prep sndr_g11_prep sndr_g12_prep sndr_g13_prep sndrd1_g11 sndrd2_g11 sndrd3_g11 sndrd4_g11 sndrd1_g12 sndrd2_g12 sndrd3_g12 sndrd4_g12 sndrd1_g13 sndrd2_g13 sndrd3_g13 sndrd4_g13 hirs3_n15 hirs3_n16 hirs3_n17 amsua_n15 amsua_n16 amsua_n17 amsub_n15 amsub_n16 amsub_n17 hsb_aqua airs_aqua amsua_aqua imgr_g08 imgr_g11 imgr_g12 pcp_ssmi_dmsp pcp_tmi_trmm sbuv2_n16 sbuv2_n17 sbuv2_n18 omi_aura ssmi_f13 ssmi_f14 ssmi_f15 hirs4_n18 hirs4_metop-a amsua_n18 amsua_metop-a mhs_n18 mhs_metop-a amsre_low_aqua amsre_mid_aqua amsre_hig_aqua ssmis_las_f16 ssmis_uas_f16 ssmis_img_f16 ssmis_env_f16 iasi_metop-a"
+   for type in $listall; do
+      count=$(ls pe*.${type}_${loop} | wc -l)
+      if [[ $count -gt 0 ]]; then
+         $(cat pe*.${type}_${loop} > diag_${type}_${string}.${YYYYMMDDHH}_mem$memid)
+      fi
+   done
+fi
+
+if [ $netcdf_diag = ".true." ]; then
+   listallnc="conv_ps conv_q conv_t conv_uv"
+
+   cat_exec="${EXECDIR}/ncdiag_cat.x"
+
+   if [ -f $cat_exec ]; then
+      print_info_msg "$VERBOSE" "
+        Copying the ncdiag_cat executable to the run directory..."
+      cp_vrfy ${cat_exec} ${analworkdir}/ncdiag_cat.x
+   else
+      print_err_msg_exit "\
+        The ncdiag_cat executable specified in cat_exec does not exist:
+        cat_exec = \"$cat_exec\"
+        Build GSI and rerun."
+   fi
+
+   for type in $listallnc; do
+      count=$(ls pe*.${type}_${loop}.nc4 | wc -l)
+      if [[ $count -gt 0 ]]; then
+         ./ncdiag_cat.x -o ncdiag_${type}_${string}.nc4.${YYYYMMDDHH}_mem$memid pe*.${type}_${loop}.nc4
+      fi
+   done
+fi
+
+done
+#
+#-----------------------------------------------------------------------
+#
+
+# next member
+   (( ensmem += 1 ))
+
+ done
+
+fi
```

Comment:
Line 761-831 should follow the same logic as line 659-705

Reply:
Thanks much for your comments! Please review the new changes and let me know if you have further questions.